lib/jekyll/regenerator.rb

```diff
@@ -104,7 +104,10 @@ def modified?(path)
     def add_dependency(path, dependency)
       return if (metadata[path].nil? || @disabled)
 
-      metadata[path]["deps"] << dependency unless metadata[path]["deps"].include? dependency
+      if !metadata[path]["deps"].include? dependency
+        metadata[path]["deps"] << dependency
+        add(dependency) unless metadata.include?(dependency)
```

Comment:
Ideally it should be. I think I ended up doing things like the `unless metadata.include?(dependency)` above, which isn't really good.